packages/internal/bridge/bridge-sample-app/src/deposit.ts

```diff
@@ -38,22 +38,24 @@ async function deposit() {
   const rootBridge: ethers.Contract = getContract("RootERC20BridgeFlowRate", params.rootBridgeAddress, params.rootProvider);
   const childBridge: ethers.Contract = getContract("ChildERC20Bridge", params.childBridgeAddress, params.childProvider);
 
-  let rootBridgeChildAddress = await rootBridge.rootTokenToChildToken(params.rootToken);
-  let childBridgeChildAddress = await childBridge.rootTokenToChildToken(params.rootToken);
-
-  if (rootBridgeChildAddress === ethers.constants.AddressZero) {
-    console.log('token not mapped, please map token before depositing');
-    return;
-  }
-
-  if (childBridgeChildAddress === ethers.constants.AddressZero) {
-    console.log('token mapping incomplete, please wait for token to map to childBridge before depositing');
-    return;
-  }
-
-  if (rootBridgeChildAddress !== childBridgeChildAddress) {
-    console.log(`token mappings mismatch on rootBridge (${rootBridgeChildAddress}) & childBridge (${childBridgeChildAddress}).`, );
-    return;
+  if (params.childToken.toUpperCase() !== 'NATIVE' && params.rootToken.toUpperCase() !== 'NATIVE') {
+    let rootBridgeChildAddress = await rootBridge.rootTokenToChildToken(params.rootToken);
+    let childBridgeChildAddress = await childBridge.rootTokenToChildToken(params.rootToken);
+    
+    if (rootBridgeChildAddress === ethers.constants.AddressZero) {
+      console.log('token not mapped, please map token before depositing');
+      return;
+    }
+  
+    if (childBridgeChildAddress === ethers.constants.AddressZero) {
+      console.log('token mapping incomplete, please wait for token to map to childBridge before depositing');
+      return;
+    }
+  
+    if (rootBridgeChildAddress !== childBridgeChildAddress) {
```

Comment:
This would signal a serious issue with the bridge. Any way we can more clearly surface this error? (see other comment)

Reply:
not sure what else to do, perhaps some of these scripts could be run as part of CI/CD or some kind of ongoing monitoring system?